### PR TITLE
CEDS-1912 Refactor summary controllers for consistency

### PR DESCRIPTION
**Pull request by SteveSugden**
This is a POC PR for the AssociateUcr journey.   Looks like more changes than there are really as I've also renamed the 2 existing controllers.  The important bits are

- AssociateUcrConfirmationController which attempts to extract the confirmation data from Flash, redirecting to Choice page if it fails (i.e. user has refreshed confirmation page)

- AssociateUcrSummaryController which now redirects to AssociateUcrConfirmationController after submission (rather than rendering the confirmation screen)

### CODE REVIEW

File: app/controllers/consolidations/AssociateUcrConfirmationController.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  authenticate: AuthenticatedAction,
+  mcc: MessagesControllerComponents,
+  associateUCRConfirmPage: associate_ucr_confirmation
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def display: Action[AnyContent] = authenticate { implicit request =>
+    (request2flash.get(FlashKeys.CONSOLIDATION_KIND), request2flash.get(FlashKeys.UCR)) match {
+      case (Some(kind), Some(ucr)) => Ok(associateUCRConfirmPage(kind, ucr))
+      case _                       => Redirect(controllers.routes.ChoiceController.displayPage)
```
Comment (PatrykRudnicki) [inline app/controllers/consolidations/AssociateUcrConfirmationController.scala]: Instead of manual redirect I'll throw an ReturnToStartPage exception

Comment (SteveSugden) [reply app/controllers/consolidations/AssociateUcrConfirmationController.scala]: That leaves the user looking at a "Something went wrong" screen.
Unless there is a plan to make throwing that exception actually "return to start page"

Comment (PatrykRudnicki) [reply app/controllers/consolidations/AssociateUcrConfirmationController.scala]: Return to start page redirect to Choice page. Is it works in different way now?

Comment (ed0906) [reply app/controllers/consolidations/AssociateUcrConfirmationController.scala]: ReturnToStartException *should* redirect you to Choice. We just need to add that catch & redirect logic into the error handler

File: app/controllers/consolidations/AssociateUcrConfirmationController.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+class AssociateUcrConfirmationController @Inject()(
+  authenticate: AuthenticatedAction,
+  mcc: MessagesControllerComponents,
+  associateUCRConfirmPage: associate_ucr_confirmation
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def display: Action[AnyContent] = authenticate { implicit request =>
+    (request2flash.get(FlashKeys.CONSOLIDATION_KIND), request2flash.get(FlashKeys.UCR)) match {
```
Comment (ed0906) [inline app/controllers/consolidations/AssociateUcrConfirmationController.scala]: Do we need to change the other Confirmation Controllers to throw ReturnToStart if the Flash data is missing?
Or have they been done already?

Comment (SteveSugden) [reply app/controllers/consolidations/AssociateUcrConfirmationController.scala]: From memory they just display "-" when the data is missing.  This is kind of the aim of this ticket - to get a consistent approach.

**Pull request closed without merging**